statsd/client.py

```diff
@@ -138,14 +151,70 @@ def _send(self, data):
             # No time for love, Dr. Jones!
             pass
 
+    def pipeline(self):
+        return Pipeline(self)
+
+
+class TCPStatsClient(StatsClientBase):
+    """TCP version of StatsClient."""
+
+    def __init__(self, host='localhost', port=8125, prefix=None, timeout=None):
+        """Create a new client."""
+        self._host = host
+        self._port = port
+        self._timeout = timeout
+        self._prefix = prefix
+        self._tlocal = threading.local()
```

Comment:
This seems straightforward but I'm wary of the potential issues by attempting to be threadsafe at all. I'd rather we just skipped this and said "TCP clients aren't threadsafe, don't share them."